netlogo-core/src/main/prim/_homedirectory.scala

```diff
@@ -0,0 +1,23 @@
+// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo
+
+package org.nlogo.prim
+
+import org.nlogo.nvm.{ Context, Reporter, RuntimePrimitiveException }
+
+class _homedirectory extends Reporter {
+
+  override def report(context: Context) =
+    report_1(context)
+  def report_1(context: Context) =
+       try {
+          System.getProperty("user.home")
```

Comment:
Can of worms.  I don't think an empty  value can occur "naturally." However a user can change it in a shell to affect all programs launched from there.

You can change your home directory to "garbage" via `export _JAVA_OPTIONS=-Duser.home=garbage`

The system will give the following error message whenever this value is accessed"
`[error] Picked up _JAVA_OPTIONS: -Duser.home=garbage`
If you then run NetLogo from sbt this value would be returned by home-directory. This would also happen with NetLogo_Console.

In this case `pathdir:get-home-path` returns "/Users/aab1638/NetLogo/netlogo-gui/garbage"
which is obviously not very helpful.

I'm willing to make whatever changes you think are reasonable. 
My thought: Make an error occur if any of the following hold

- the path is relative, not absolute
- the directory does not exist

Reply:
Yeah, probably not worth worrying about, we can let the user check with `file-exists?` if they need to confirm it returns a good value.